test/timeouts.js

```diff
@@ -36,6 +37,73 @@ describe('Timeout and max received events for subscriptions', function () {
     nsc.stopServer(server, done)
   })
 
+  // connect to a server that never sends data
+  it('conn timeout - socket timeout', (done) => {
+    const srv = net.createServer((c) => {})
+    srv.listen(0, () => {
+      const p = srv.address().port
+      const nc = NATS.connect({ port: p, timeout: 1000 })
+      nc.on('connect', () => {
+        done(new Error('should have failed'))
+      })
+      nc.on('error', (err) => {
+        err.should.be.instanceof(NATS.NatsError)
+        err.chainedError.should.be.instanceof(NATS.NatsError)
+        err.chainedError.should.have.property('code', NATS.CONN_TIMEOUT)
+        srv.close(done)
+      })
+    })
+  }, 15000)
+
+  it('connection timeout - fail to connect quickly enough', (done) => {
+    // this is not expecting to have enough time to connect
+    const nc = NATS.connect('nats://connect.ngs.global', { timeout: 1 })
+    nc.on('error', (err) => {
+      err.should.be.instanceof(NATS.NatsError)
+      err.chainedError.should.be.instanceof(NATS.NatsError)
+      err.chainedError.should.have.property('code', NATS.CONN_TIMEOUT)
+      done()
+    })
+
+    nc.on('connect', () => {
+      done(new Error('should have failed'))
+    })
+  })
+
+  // connect to a server that never sends data
+  it('conn timeout - reconnects work', (done) => {
+    const srv = net.createServer((c) => {})
+    srv.listen(0, () => {
+      const p = srv.address().port
+      const nc = NATS.connect('nats://localhost:' + p, {
+        timeout: 1000,
+        servers: ['nats://localhost:' + PORT]
+      })
+      nc.on('connect', () => {
+        nc.currentServer.url.host.should.be.equal('localhost:' + PORT)
+        nc.close()
+        srv.close(done)
+      })
+      nc.on('error', (err) => {
+        done(new Error('shouldnt have failed: ' + err))
+      })
+    })
+  }, 15000)
+
+  it('connection timeout doesnt trigger', (done) => {
+    // this is not expecting to have enough time to connect
```

Comment:
I think this comment is wrong? You are expecting the connect to have enough time and check that no connection timeout error is raised, right?

Reply:
I am expecting that in 1ms, it is not going to connect to the remote system, so it triggers the first condition.

Reply:
No, this test passes 1000 (1sec) and I think expect to be able to connect and make sure that this is not raising the timeout error..

Reply:
c/p it is the lower function.